content/posts/2025-01-30-ebpf-flows-filtering.md

```diff
@@ -0,0 +1,225 @@
+
+Note:
+
+- You can't use ports and either sourcePorts or destPorts in the same rule.
+
+The Following configuration example shows some of the possible filtering options with multi rules filters
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 10.128.0.0/24
+          peerCIDR: 10.129.0.0/24
+          ports: 443,6443
+          protocol: TCP
+          sampling: 10
+        - action: Accept
+          cidr: 10.129.0.1/24
+          ports: 53
+          protocol: UDP
+          sampling: 20
+        - action: Reject
+          tcpFlags: "SYN"
+          cidr: 10.130.0.0/24
+          protocol: TCP
+          sourcePorts: 80-100
+        - action: Accept
+          cidr: 172.30.0.0/16
+          protocol: SCTP
+          pktDrops: true
+        - action: Reject
+          cidr: 8.8.8.8/32
+          protocol: ICMP
+          icmpType: 8 // ICMP Echo request packet
+```
+2️⃣ Packet Inspection – Extract relevant packet attributes within an eBPF program.
+
+3️⃣ Early Flow Filtering – Drop or pass packets based on predefined conditions before sending data to user space.
+
+The Following Diagram shows how eBPF filtering is done
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-flows-filtering.png')}" alt="eBPF Flows Filtering Processing">
+
+Note:
+- The maximum number of eBPF flow filter rules is limited to 16 to ensure conservative memory
+usage when this feature is enabled.
+
+- rules with duplicate CIDRs are rejected via validation webhook.
+
+- Both IPv4 and IPv6 formats are supported.
+
+- In case users wanted to match on any CIDR Null CIDR can be used like `0.0.0.0/0` which is also the default.
+
+## Key Use Cases
+
+🚀 Reducing Observability Overhead – Avoid logging irrelevant flows in high-traffic Kubernetes clusters.
+
+🔐 Security Filtering – Focus on anomalous or suspicious traffic while ignoring normal flows.
+
+🌐 Network Performance Monitoring – Capture only high-latency or dropped-packet flows for troubleshooting.
+
+### Filter EastWest and NorthSouth flows
+
+Let's allow serviceIP 172.30.100.64:80 with sample 10
+and allow pods traffic between subnet 10.128.0.0/16 and 10.129.0.0/16 with sampling 20 and reject everything else
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          sampling: 10
+        - action: Accept
+          cidr: 10.128.0.0/16
+          peerCIDR: 10.129.0.0/16
+          sampling: 20
+        - action: Reject
+          cidr: 0.0.0.0/0
+``` 
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-svc-and-pods-flows.png')}" alt="eBPF Flows Filtering Kubernetes NorthSouth and EastWest Flows">
+
+### Filter flows with packet drops
+
+let's filter any kubernetes service flows with a packet drop and reject everything else
```

Comment:
Add Note for this saying PacketDrop feature needs to be enabled with eBPF in privileged mode